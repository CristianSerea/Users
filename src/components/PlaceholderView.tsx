import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {fonts} from '../constants/fonts';

type Props = {
  text: string;
};

const PlaceholderView = ({text}: Props) => {
  return (
    <View style={styles.container} testID="PlaceholderView">
      <Text style={fonts.normal} testID={text}>
        {text}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default PlaceholderView;
