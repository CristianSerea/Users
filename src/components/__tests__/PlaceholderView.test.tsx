import React from 'react';
import '@testing-library/jest-native/extend-expect';
import {render} from '@testing-library/react-native';
import PlaceholderView from '../PlaceholderView';

describe('PlaceholderView tests', () => {
  it('Test placeholder view rendering', () => {
    const text = 'Test';
    const {getByTestId, toJSON} = render(<PlaceholderView text={text} />);

    expect(getByTestId('PlaceholderView')).toBeTruthy();
    expect(getByTestId(`${text}`)).toHaveTextContent(text);

    expect(toJSON()).toMatchSnapshot();
  });
});
