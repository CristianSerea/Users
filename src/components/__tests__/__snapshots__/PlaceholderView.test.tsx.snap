// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderView tests Renders placeholder view 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="PlaceholderView"
>
  <Text
    style={
      {
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
    testID="Test"
  >
    Test
  </Text>
</View>
`;

exports[`PlaceholderView tests Test placeholder view rendering 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="PlaceholderView"
>
  <Text
    style={
      {
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
    testID="Test"
  >
    Test
  </Text>
</View>
`;
